src/bin/sse-test-api/main.rs

```diff
@@ -0,0 +1,195 @@
+mod stream_entity;
+
+use actix_web::{guard, web, App, HttpRequest, HttpResponse, HttpServer, Responder};
+use eventsource_client as es;
+use futures::executor;
+use serde::{self, Deserialize, Serialize};
+use std::collections::HashMap;
+use std::sync::{mpsc, Mutex};
+use std::thread;
+use stream_entity::StreamEntity;
+
+#[derive(Serialize)]
+struct Status {
+    capabilities: Vec<String>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct Config {
+    /// The URL of an SSE endpoint created by the test harness.
+    stream_url: String,
+    /// The URL of a callback endpoint created by the test harness .
+    callback_url: String,
+    /// An optional integer specifying the initial reconnection delay parameter, in
+    /// milliseconds. Not all SSE client implementations allow this to be configured, but the
+    /// test harness will send a value anyway in an attempt to avoid having reconnection tests
+    /// run unnecessarily slowly.
+    initial_delay_ms: Option<u64>,
+    /// An optional integer specifying the read timeout for the connection, in
+    /// milliseconds.
+    read_timeout_ms: Option<u64>,
+    /// A JSON object containing additional HTTP header names and string values. The SSE
+    /// client should be configured to add these headers to its HTTP requests; the test harness
+    /// will then verify that it receives those headers. The test harness will only set this
+    /// property if the test service has the "headers" capability. Header names can be assumed
+    /// to all be lowercase.
+    headers: Option<HashMap<String, String>>,
+}
+
+#[derive(Serialize, Debug)]
+#[serde(tag = "kind")]
+enum EventType {
+    #[serde(rename = "event")]
+    Event { event: Event },
+    #[serde(rename = "error")]
+    Error { error: String },
+}
+
+impl From<es::Event> for EventType {
+    fn from(event: es::Event) -> Self {
+        Self::Event {
+            event: Event {
+                event_type: event.event_type.clone(),
+                data: String::from_utf8(event.field("data").unwrap_or_default().to_vec()).unwrap(),
+                id: String::from_utf8(event.field("id").unwrap_or_default().to_vec()).unwrap(),
+            },
+        }
+    }
+}
+
+#[derive(Serialize, Debug)]
+struct Event {
+    #[serde(rename = "type")]
+    event_type: String,
+    data: String,
+    id: String,
+}
+
+async fn status() -> impl Responder {
+    web::Json(Status {
+        capabilities: vec![
+            "comments".to_string(),
+            "retry".to_string(),
```

Comment:
You can remove retry. I added that when I was trying to make a change to the SSE test harness. That didn't get merged in (related to the discussion we all had recently about default behavior).